### PR TITLE
Future integrate sla

### LINKED ISSUE #8371: Integrate the new liquidity SLA with futures and existing order types
As the new liquidity mechanism (Liquidity SLA) will replace the old liquidity mechanism we need to integrate sla liquidity into futures and existing order types and remove all the old liquidity mechanism.

### PR DESCRIPTION
**Pull request by karlem**
close #8371 .

### CODE REVIEW

File: core/execution/future/liquidity_provision.go [hunk 1]
```diff
@@ -35,998 +26,33 @@ func (m *Market) SubmitLiquidityProvision(
...
-) {
+) error {
 	defer m.onTxProcessed()
 
-	m.idgen = idgeneration.New(deterministicID)
-	defer func() { m.idgen = nil }()
-
-	if !m.canSubmitCommitment() {
-		return common.ErrCommitmentSubmissionNotAllowed
-	}
```
Comment (jeremyletang) [inline core/execution/future/liquidity_provision.go:47]: we probably still want this.

Comment (jeremyletang) [reply core/execution/future/liquidity_provision.go:47]: unless it's done in the liquidity engine?

File: core/execution/future/market.go [hunk 2]
```diff
@@ -896,7 +881,12 @@ func (m *Market) closeMarket(ctx context.Context, t time.Time, finalState types.
 	}
 
 	// final distribution of liquidity fees
-	m.distributeLiquidityFees(ctx)
+	// TODO karel - consider having one function for this instead of calling both AllocateFees and OnEpochEnd
+	if err := m.liquidity.AllocateFees(ctx); err != nil {
+		log.Panic(err)
```
Comment (jeremyletang) [inline core/execution/future/market.go:886]: if we keep the panic, better use the logger in the Market instead.

File: core/execution/future/market.go [hunk 3]
```diff
@@ -1075,9 +1059,6 @@ func (m *Market) enterAuction(ctx context.Context) {
 		updatedOrders = append(updatedOrders, order)
 	}
 
-	// now update all special orders
-	m.enterAuctionSpecialOrders(ctx, updatedOrders)
```
Comment (jeremyletang) [inline core/execution/future/market.go:1079]: This will still be needed for ther pegged orders.

File: core/execution/future/market.go [hunk 4]
```diff
@@ -1182,7 +1163,6 @@ func (m *Market) leaveAuction(ctx context.Context, now time.Time) {
 	// set the mark price here so that margins checks for special orders use the correct value
 	m.markPrice = m.getLastTradedPrice()
 
-	m.checkForReferenceMoves(ctx, updatedOrders, true)
```
Comment (jeremyletang) [inline core/execution/future/market.go:1185]: This i still needed for the pegged orders.

File: core/execution/future/market.go [hunk 5]
```diff
@@ -1674,11 +1652,6 @@ func (m *Market) SubmitOrderWithIDGeneratorAndOrderID(
 		[]*types.Order{conf.Order}, conf.PassiveOrdersAffected...)
 	allUpdatedOrders = append(allUpdatedOrders, orderUpdates...)
 
-	if !m.as.InAuction() {
-		m.checkForReferenceMoves(
-			ctx, allUpdatedOrders, false)
-	}
-
```
Comment (jeremyletang) [inline core/execution/future/market.go:1681]: same here.

File: core/execution/future/liquidity_provision.go (hunk 1 above)
Comment (karlem) [reply core/execution/future/liquidity_provision.go:47]: Correct. It is in the market liquidity

File: core/execution/future/special_orders.go [hunk 6]
```diff
@@ -142,207 +139,38 @@ func (m *Market) repriceAllSpecialOrders(
...
-
-func (m *Market) stopAllSpecialOrders(
-	ctx context.Context,
-	updatedOrders []*types.Order,
-) {
-	// Park all pegged orders
-	updatedOrders = append(
-		updatedOrders,
-		m.parkAllPeggedOrders(ctx)...,
-	)
```
Comment (jeremyletang) [inline core/execution/future/special_orders.go:219]: some form of this function is still needed for pegged orders

File: core/execution/future/special_orders.go [hunk 7]
```diff
@@ -373,99 +201,14 @@ func (m *Market) updateMargins(ctx context.Context, partiesPos map[string]events
 	return m.updateMargin(ctx, positions), positions, marginsBefore
 }
 
-func (m *Market) applyBondPenaltiesAndCancelLPs(
+func (m *Market) enterAuctionSpecialOrders(
```
Comment (jeremyletang) [inline core/execution/future/special_orders.go:204]: updatedOrders can be removed here.

File: core/execution/future/special_orders.go [hunk 8]
```diff
@@ -123,13 +122,11 @@ func (m *Market) repriceAllSpecialOrders(
 	ctx context.Context,
 	changes uint8,
 	orderUpdates []*types.Order,
```
Comment (jeremyletang) [inline core/execution/future/special_orders.go:124]: orderUpdates can be removed here. we also probably don't need to return anything anymore

File: core/execution/future/special_orders.go (hunk 6 above)
Comment (karlem) [reply core/execution/future/special_orders.go:219]: It's the `parkAllPeggedOrders` function. It's called elsewhere.

File: core/execution/future/special_orders.go (hunk 8 above)
Comment (karlem) [reply core/execution/future/special_orders.go:124]: orderUpdates are for the same thing as before. For the updates - don't we want to change the prices again after the resubmitting of the pegged orders?

**Comment by jeremyletang** [comment]
closed in favour of this PR (https://github.com/vegaprotocol/vega/pull/8949) being rebased along the way.

**Pull request closed without merging**